src/mds/objects.py

```diff
@@ -0,0 +1,236 @@
+    """
+    Create object.
+
+    file_name (str): Name for the file being uploaded
+    authz (dict): authorization block with requirements for what's being uploaded
+    aliases (list, optional): unique name to allow using in place of whatever GUID gets
+        created for this upload
+    metadata (dict, optional): any additional metadata to attach to the upload
+    """
+
+    file_name: str
+    authz: dict
+    aliases: list = None
+    metadata: dict = None
+
+
+@mod.post("/objects")
+async def create_object(
+    body: CreateObjInput,
+    request: Request,
+    token: HTTPAuthorizationCredentials = Security(bearer),
+):
+    """
+    Create object placeholder and attach metadata, return Upload url to the user.
+
+    Args:
+        body (CreateObjInput): input body for create object
+    """
+    try:
+        # NOTE: token can be None if no Authorization header was provided, we expect
+        #       this to cause a downstream exception since it is invalid
+        token_claims = await access_token("user", "openid", purpose="access")(token)
+    except Exception as exc:
+        logger.error(exc, exc_info=True)
+        raise HTTPException(
+            HTTP_401_UNAUTHORIZED,
+            f"Could not verify, parse, and/or validate scope from provided access token.",
+        )
+
+    file_name = body.dict().get("file_name")
+    authz = body.dict().get("authz")
+    aliases = body.dict().get("aliases") or []
+    metadata = body.dict().get("metadata")
+    logger.debug(f"validating authz block input: {authz}")
+
+    if not _is_authz_version_supported(authz):
+        raise HTTPException(HTTP_400_BAD_REQUEST, f"Unsupported authz version: {authz}")
+
+    logger.debug(f"validated authz.resource_paths: {authz.get('resource_paths')}")
+
+    if not isinstance(authz.get("resource_paths"), Iterable):
+        raise HTTPException(
+            HTTP_400_BAD_REQUEST,
+            f"Invalid authz.resource_paths, must be valid list of resources, got: {authz.get('resource_paths')}",
+        )
+
+    metadata = metadata or {}
+
+    # get user id from token claims
+    uploader = token_claims.get("sub")
+    auth_header = str(request.headers.get("Authorization", ""))
+
+    blank_guid, signed_upload_url = await _create_blank_record_and_url(
+        file_name, authz, auth_header
+    )
+
+    if aliases:
+        await _create_aliases_for_record(aliases, blank_guid, auth_header)
+
+    await _add_metadata(blank_guid, metadata, authz, uploader)
+
+    response = {
+        "guid": blank_guid,
+        "aliases": aliases,
+        "metadata": metadata,
+        "upload_url": signed_upload_url,
+    }
+
+    return JSONResponse(response, HTTP_201_CREATED)
+
+
+async def _create_aliases_for_record(aliases: list, blank_guid: str, auth_header: str):
+    aliases_data = {"aliases": [{"value": alias} for alias in aliases]}
+    logger.debug(f"trying to create aliases: {aliases_data}")
+    try:
+        async with httpx.AsyncClient() as client:
```

Comment:
sorry for this very late decision - we might want to sharer a single HTTP client instance across the application: https://www.python-httpx.org/advanced/#why-use-a-client

Reply:
ah, interesting, is it okay to leave the connection open as along as the application is running? are you thinking we setup the client during app creation like

```
async_client = httpx.AsyncClient()
```

and then just import and use it everywhere without an explicit `.close()`?

Reply:
Right, I think it is okay to do that. To be clean, we may want to call `close()` on application shutdown.